logger/logger.go

```diff
@@ -150,3 +155,19 @@ func getLoggers() map[string]Logger {
 
 	return l
 }
+
+// NewContext returns a new Context, derived from ctx, which carries the
+// provided Logger.
+func NewContext(ctx context.Context, logger Logger) context.Context {
+	return context.WithValue(ctx, contextKey{}, logger)
```

Comment:
I believe it's best to use a constant value rather than re-creating it every time. So:

```go
type logContextKeyType struct{}

// logContextKey is how we find Loggers in a context.Context
var logContextKey = logContextKeyType{}

// ... down below

return context.WithValue(ctx, logContextKey, logger)
```